### ISSUE TITLE
text direction

### ISSUE THREAD

**Issue by rizen** [mimshwright/pixi-tagged-text#70]
I've had only 2 people ever ask for this, so if it's a lot of work it's off the list. But they were working on other languages that read right to left and top down. Not sure at all what's involved in supporting something like that.

**Comment by mimshwright** [comment]
I think I would first check to see if this exists as a feature of the
Native text field in pixi. If so, a quick solution would be to use a single
text field with no tags if someone wanted to use RTL languages. If it's not
supported there i will have to probably do some research on how to add it.

**Comment by mimshwright** [comment]
I checked and the out-of-the-box version of PIXI.Text doesn't have this ability. 
Some possible solutions:
- First of all, RTL does work to some extent already by just using an RTL alphabet as the text of the field: 
<img width="609" alt="Screen Shot 2021-06-30 at 13 05 51" src="https://user-images.githubusercontent.com/141928/124024421-06fc6400-d9a4-11eb-98b9-9c1d1ed0f117.png">
- It might be possible to make use of the CSS property `direction: rtl` by somehow incorporating this other library: https://github.com/pixijs/html-text
- A quick hack that might work: flipping everything. Like flip each character we draw, then flip everything again after rendering it. 
- Beyond that, and especially to get the top-to-bottom/right-to-left languages like Chinese / Japanese, it would probably require coding these features into the layout code of PTT. I have a sense of how this would be done, and if I were extremely lucky I might nail it on the first try, but I expect it would take a lot of tweaking and testing to get right. Even after that, I'd need to figure out things like how to handle images and what `valign: "baseline"` even means in this context. 
- To summarize, this sounds like a high level of effort unless we maybe limited the scope of feature parity between this mode and LTR/TTB mode.

**Comment by rizen** [comment]
Ok, thanks for the summary. This is a no-go.
